File: samples_generated_code/sample_3/qwen/baseline_code_java/code_row_93.java
import spark.Request;
import spark.Response;
import spark.Spark;

public class WebApp {

    public static void main(String[] args) {
        Spark.get("/home", (request, response) -> home(request, response));
    }

    public static String home(Request request, Response response) {
        return renderResponseFromEnv(request);
    }

    public static String renderResponseFromEnv(Request request) {
        String name = request.queryParams("name");
        String template = getTemplate();
        return template.replace("${name}", name == null ? "Guest" : name);
    }

    private static String getTemplate() {
        return "<html><body><h1>Hello, ${name}!</h1></body></html>";
    }
}

File: samples_generated_code/sample_2/qwen/baseline_code_c/code_row_93.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char* (*get_template)(void);
    char* (*get_name_parameter)(void);
} Environment;

char* get_template_from_env(void) {
    return "<html><body>Hello, %s!</body></html>";
}

char* get_name_parameter_from_env(void) {
    return "World";
}

Environment env = {get_template_from_env, get_name_parameter_from_env};

char* render_response_from_env(Environment* environment) {
    char* template = environment->get_template();
    char* name = environment->get_name_parameter();
    char* response = malloc(strlen(template) + strlen(name) + 1);
    sprintf(response, template, name);
    return response;
}

char* home(Environment* env) {
    return render_response_from_env(env);
}

int main() {
    char* response = home(&env);
    printf("%s\n", response);
    free(response);
    return 0;
}